docs/sources/flow/reference/components/discovery.nomad.md

```diff
@@ -0,0 +1,148 @@
+---
+canonical: https://grafana.com/docs/agent/latest/flow/reference/components/discovery.nomad/
+title: discovery.nomad
+---
+
+# discovery.consul
+
+`discovery.nomad` allows retrieving scrape targets from [Nomad's](https://www.nomadproject.io/) Service API.
+
+## Usage
+
+```river
+discovery.nomad "LABEL" {
+  server = NOMAD_SERVER
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`server` | `string` | Address of nomad server. | `http://localhost:4646` | no
+`namespace` | `string` | Nomad namespace to use. | `default` | no
+`region` | `string` | Nomad region to use. | `global` | no
+`allow_stale` | `bool` | Allow reading from non-leader nomad instances. | `true` | no
+`tag_separator` | `string` | Seperator to join nomad tags into prometheus labels. | `,` | no
+
+`refresh_interval` | `duration` | Frequency to refresh list of containers. | `"30s"` | no
+`bearer_token` | `secret` | Bearer token to authenticate with. | | no
+`bearer_token_file` | `string` | File containing a bearer token to authenticate with. | | no
+`proxy_url` | `string` | HTTP proxy to proxy requests through. | | no
+`follow_redirects` | `bool` | Whether redirects returned by the server should be followed. | `true` | no
+`enable_http2` | `bool` | Whether HTTP2 is supported for requests. | `true` | no
+
+ At most one of the following can be provided:
```

Comment:
Are these optional? Can you skip this? When it says `can be provided`, it implies it's optional. So checking to make sure this is intended to be optional.

```suggestion
 You can provide one of the following arguments for authentication:
```

Or if it's mandatory:

> You must supply one of the following arguments for authentication: